renku/core/dataset/providers/api.py

```diff
@@ -215,3 +234,54 @@ def get_access_token_url(self) -> str:
     def export(self, **kwargs) -> str:
         """Execute export process."""
         raise NotImplementedError
+
+
+class ProviderCredentials(abc.ABC, UserDict):
+    """Credentials of a provider.
+
+    NOTE: A ``None`` value for a key means that it is not set; however, an empty value, "", means that the key was set.
+    """
+
+    def __init__(self, provider: ProviderApi):
+        super().__init__()
+        self._provider: ProviderApi = provider
+        self.data: Dict[str, Optional[str]] = {key: None for key in self.get_canonical_credentials_names()}
+
+    @staticmethod
+    @abc.abstractmethod
+    def get_credentials_names() -> Tuple[str, ...]:
```

Comment:
Should this just return `List[str]` as per the docstring?

Reply:
This won't change and makes sense to be a tuple. I've updated the docstring.